cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/Query.java

```diff
@@ -53,77 +53,71 @@
+     * index is up to date.
      * </p>
      * <p>
-     * The default options are {@code indexType=JSON, tokenize=null}.
+     * Note that the sort order on fields for index definitions currently only supports the
+     * {@link com.cloudant.sync.query.FieldSort.Direction#ASCENDING} direction. If there are one or
+     * more members of {@code fieldNames} with a direction of
+     * {@link com.cloudant.sync.query.FieldSort.Direction#DESCENDING}, an exception will be thrown.
      * </p>
      * <p>
-     * If an index with these fields and options already exists, then no new index is created, and
-     * the existing equivalent index name is returned.
+     * If an index with these fields already exists, then no new index is created, and the
+     * existing equivalent {@link Index} is returned. Note that the existing equivalent index
+     * may have a different name to the one requested.
      * </p>
-     * @param fieldNames the fields to index
-     * @return the generated index name, or the existing index name (see above)
-     * @throws QueryException if there was a problem creating the index
-     */
-    String ensureIndexed(List<FieldSort> fieldNames) throws QueryException;
-
-    /**
      * <p>
-     * Create an {@link Index} with a given name and default options.
+     * Whether the index previously existed or not, this method will ensure the index is up to date.
      * </p>
      * <p>
-     * The default options are {@code indexType=JSON, tokenize=null}.
-     * </p>
-     * <p>
-     * If an index with these fields and options already exists, then no new index is created, and
-     * the existing equivalent index name is returned.
+     * For new indexes, the index data will be created for the first time. For existing indexes, the
+     * index data will be updated by reading all documents in the
+     * {@link com.cloudant.sync.documentstore.DocumentStore} which have changed since the last
+     * update.
      * </p>
+     *
      * @param fieldNames the fields to index
-     * @param indexName the name of the index to be created
-     * @return the requested index name, or the existing index name (see above)
-     * @throws QueryException if there was a problem creating the index
+     * @param indexName  the name of the index to be created, or null for an automatically
+     *                   generated name
+     * @return the requested {@link Index}, or the existing {@link Index} (see above)
+     * @throws QueryException if there was a problem creating or updating the index
      */
-    String ensureIndexed(List<FieldSort> fieldNames, String indexName) throws QueryException;
+    Index ensureIndexedJson(List<FieldSort> fieldNames, String indexName) throws QueryException;
 
     /**
      * <p>
-     * Create an {@link Index} with a given name and of a given type.
+     * Create a text {@link Index} for one or more fields with an optional index name and SQLite FTS
```

Comment:
It may be worth saying `optional` again i.e. `and optional SQLite ...`
I'm not a grammar expert, but I'm not fully convinced that optional applies to both sides of the conjunction.